integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/e2e/openstack/OpenStackTests.java

```diff
@@ -0,0 +1,228 @@
+    public void testOpenStackWithDefaulkHDFClusterDefinitionAndPrewarmedImagesWithoutGateWaySetup(
+            TestContext testContext,
+            String clusterDefinitionName,
+            String[] groups,
+            String scaleGroup,
+            @Description TestCaseDescription description) {
+
+        testContext
+                .given(ClusterTestDto.class)
+                .withClusterDefinitionName(clusterDefinitionName)
+                .given(StackTestDto.class)
+                .withInstanceGroupsEntity(getInstanceGroupTestDtos(testContext, groups))
+                .withCluster(ClusterTestDto.class.getSimpleName())
+                .when(stackTestClient.createV4())
+                .await(STACK_AVAILABLE)
+                .when(stackTestClient.scalePostV4()
+                        .withGroup(scaleGroup)
+                        .withDesiredCount(2))
+                .await(STACK_AVAILABLE)
+                .when(stackTestClient.scalePostV4()
+                        .withGroup(scaleGroup)
+                        .withDesiredCount(1))
+                .await(STACK_AVAILABLE)
+                .when(stackTestClient.stopV4())
+                .await(STACK_STOPPED)
+                .when(stackTestClient.startV4())
+                .await(STACK_AVAILABLE)
+                .then((tc, testDto, cc) -> stackTestClient.deleteV4().action(tc, testDto, cc))
+                .validate();
+    }
+
+    @DataProvider(name = "openStackHdfTestsDataProvider")
+    public Object[][] openStackHdfTestsDataProvider() {
+        SparklessTestContext testContext = getBean(SparklessTestContext.class);
+        minimalSetupForClusterCreation(testContext);
+
+        Object[][] data = new Object[getHdfClusterDefinition().length][5];
+        for (int i = 0; i < getHdfClusterDefinition().length; i++) {
+            data[i][0] = testContext;
+            data[i][1] = getHdfClusterDefinition()[i];
+            data[i][2] = getHdfGroups();
+            data[i][3] = getHdfScaleGroup();
+            data[i][4] = new TestCaseDescription.TestCaseDescriptionBuilder()
+                    .given("there is an available cloudbreak")
+                    .when("a stack create request with "
+                            + getHdfClusterDefinition()[i] + " cluster definition is sent AND scale AND stop AND start requests are sent")
+                    .then("all stack operation should succeed and the stack should be deletable");
+        }
+        if (isHdfEnabled()) {
+            return data;
+        }
+        return new Object[0][5];
+    }
+
+    private List<InstanceGroupTestDto> getInstanceGroupTestDtos(TestContext testContext, String[] groups) {
+        List<InstanceGroupTestDto> instanceGroups = new ArrayList<>();
+        for (int i = 0; i < groups.length; i++) {
+            HostGroupType hostGroupType = HostGroupType.getByName(groups[i]);
+            instanceGroups.addAll(InstanceGroupTestDto.withHostGroup(testContext, hostGroupType));
```

Comment:
`testContext.given(InstanceGroupTestDto.class).withHostGroup(...)` should be used here

Reply:
Or it should be implemented in the constructor of the `AbstractCloudbreakTestDto` to register itself in the test context. In fact this second option is favorable.